### ISSUE TITLE
Flaky Tests - RollCycleMultiThreadStressTest.stress and DocumentOrderingTest.queuedWriteInPreviousCycleShouldRespectTotalOrdering

### ISSUE THREAD

**Issue by tomshercliff** [OpenHFT/Chronicle-Queue#459]

**Comment by JerryShea** [comment]
Also http://teamcity.higherfrequencytrading.com:8111/viewLog.html?buildId=258956&tab=buildResultsDiv&buildTypeId=OpenHFT_ChronicleQueue4_Snapshot

**Comment by RobAustin** [comment]
```
[main] INFO net.openhft.chronicle.queue.DirectoryUtils - Tmp dir: /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbn
[main] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbn/directory-listing.cq4t took 25.846 ms.
[main] INFO net.openhft.chronicle.queue.DirectoryUtils - Tmp dir: /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo
[main] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/directory-listing.cq4t took 0.198 ms.

org.junit.AssumptionViolatedException: ordering/atomicity is not guaranteed when using progressOnContention = true,as multiple threads can be concurrently executing within a queue's document context when the queue head is contented


	at org.junit.Assume.assumeTrue(Assume.java:59)
	at org.junit.Assume.assumeFalse(Assume.java:66)
	at net.openhft.chronicle.queue.impl.single.DocumentOrderingTest.queuedWriteInPreviousCycleShouldRespectTotalOrdering(DocumentOrderingTest.java:62)
	at net.openhft.chronicle.queue.impl.single.DocumentOrderingTest.reapeatQueuedWriteInPreviousCycleShouldRespectTotalOrdering(DocumentOrderingTest.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)

[main] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072651.cq4 took 0.564 ms.
[pool-2-thread-2] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072651.cq4 took 0.061 ms.
[pool-2-thread-1] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072651.cq4
[pool-2-thread-2] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072653.cq4 took 0.931 ms.
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts - Rolled 2 times to find the next cycle file. This can occur if your appenders have not written anything for a while, leaving the cycle files with a gap.
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072651.cq4
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072651.cq4
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072653.cq4
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072653.cq4
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbo/20180612-072653.cq4
[main] INFO net.openhft.chronicle.queue.DirectoryUtils - Tmp dir: /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp
[main] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/directory-listing.cq4t took 0.163 ms.
[main] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/20180612-072653.cq4 took 0.224 ms.
[pool-2-thread-1] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/20180612-072653.cq4 took 0.059 ms.
[pool-2-thread-2] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/20180612-072653.cq4 took 0.05 ms.
[pool-2-thread-1] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/20180612-072655.cq4 took 0.419 ms.
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/20180612-072655.cq4
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/20180612-072653.cq4
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/20180612-072653.cq4
[main] DEBUG net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder - File released /Users/robaustin/git-projects/Chronicle-Queue/target/document-ordering-jibd5bbp/20180612-072655.cq4

java.lang.AssertionError: 
Expected :0
Actual   :3
 <Click to see difference>
```

**Comment by RobAustin** [comment]
this relates to https://github.com/OpenHFT/Chronicle-Queue/issues/498

**Comment by dpisklov** [comment]
Fixed by #504

**Comment by hft-team-city** [comment]
Released in Chronicle-Queue-5.16.0 (https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.16.0), BOM-1.16.74 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-1.16.74)
